config/src/main/java/com/typesafe/config/Config.java

```diff
@@ -794,6 +796,44 @@
     Duration getDuration(String path);
 
     /**
+     * Gets a value as a java.time.Period. If the value is
+     * already a number, then it's taken as days; if it's
+     * a string, it's parsed understanding units suffixes like
+     * "10d" or "5w" as documented in the <a
+     * href="https://github.com/typesafehub/config/blob/master/HOCON.md">the
+     * spec</a>. This method never returns null.
+     *
+     * @since 1.3.0
+     *
+     * @param path
+     *            path expression
+     * @return the period value at the requested path
+     * @throws ConfigException.Missing
+     *             if value is absent or null
+     * @throws ConfigException.WrongType
+     *             if value is not convertible to Long or String
+     * @throws ConfigException.BadValue
+     *             if value cannot be parsed as a number of the given TimeUnit
+     */
+    Period getPeriod(String path);
+
+    /**
+     * Gets a value as a java.time.temporal.TemporalAmount.
+     * This method will first try get get the value as a java.time.Duration, and if unsuccessful,
+     * then as a java.time.Period.
+     * This means that values like "5m" will be parsed as 5 minutes rather than 5 months
```

Comment:
`TemporalAmount` is not really meant to be used in application code. According to the [javadoc](https://docs.oracle.com/javase/8/docs/api/java/time/temporal/TemporalAmount.html):

> This interface is a framework-level interface that should not be widely used in application code. Instead, applications should create and pass around instances of concrete types, such as Period and Duration.

If you're going to force users to decide how to handle the `TemporalAmount` returned from this method, you might as well let them implement the relatively simple logic in `getTemporal` themselves, so they can decide how to handle the ambiguity of minutes/months.

In a practical sense, I think it would be more useful to have `getDuration` support weeks, months, and years as units using standard lengths for those units. In most cases the underlying API I'm using (e.g. max-age for a cookie) expects a precise number of nanoseconds, so a `Period` is usually not what I want anyway.